shell/platform/android/image_external_texture_gl.cc

```diff
@@ -173,12 +184,42 @@ void ImageExternalTextureGLImpeller::Attach(PaintContext& context) {
 
 void ImageExternalTextureGLImpeller::ProcessFrame(PaintContext& context,
                                                   const SkRect& bounds) {
-  const bool swapped = MaybeSwapImages();
-  if (!swapped && !egl_image_.is_valid()) {
-    // Nothing to do.
+  JavaLocalRef image = AcquireLatestImage();
+  if (image.is_null()) {
     return;
   }
+  // NOTE: In the following code it is important that old_android_image is
+  // not closed until after the update of egl_image_ otherwise the image might
+  // be closed before the old EGLImage referencing it has been deleted. After
+  // an image is closed the underlying HardwareBuffer may be recycled and used
+  // for a future frame.
+  JavaLocalRef old_android_image(android_image_);
+  android_image_.Reset(image);
+  JavaLocalRef hardware_buffer = HardwareBufferFor(image);
+  AHardwareBuffer* latest_hardware_buffer = AHardwareBufferFor(hardware_buffer);
+  auto key = flutter::NDKHelpers::AHardwareBuffer_getId(latest_hardware_buffer);
+  auto existing_image = FindImage(key);
+  if (existing_image != nullptr) {
+    dl_image_ = existing_image;
+
+    CloseHardwareBuffer(hardware_buffer);
+    // IMPORTANT: We only close the old image after texture stops referencing
```

Comment:
This comment should be re-written:

// IMPORTANT: We have just received a new frame to display so close the previous Image so that it is recycled and used for a future frame.

Reply:
Done